db/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/db/river/LagreJournalpostIdRiver.kt

```diff
@@ -0,0 +1,140 @@
+package no.nav.helsearbeidsgiver.inntektsmelding.db.river
+
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.json.JsonElement
+import no.nav.helsearbeidsgiver.felles.BehovType
+import no.nav.helsearbeidsgiver.felles.EventName
+import no.nav.helsearbeidsgiver.felles.Key
+import no.nav.helsearbeidsgiver.felles.json.krev
+import no.nav.helsearbeidsgiver.felles.json.les
+import no.nav.helsearbeidsgiver.felles.json.lesOrNull
+import no.nav.helsearbeidsgiver.felles.json.toJson
+import no.nav.helsearbeidsgiver.felles.json.toPretty
+import no.nav.helsearbeidsgiver.felles.loeser.ObjectRiver
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.model.Fail
+import no.nav.helsearbeidsgiver.felles.utils.Log
+import no.nav.helsearbeidsgiver.inntektsmelding.db.AapenImRepo
+import no.nav.helsearbeidsgiver.inntektsmelding.db.InntektsmeldingRepository
+import no.nav.helsearbeidsgiver.utils.collection.mapValuesNotNull
+import no.nav.helsearbeidsgiver.utils.json.fromJson
+import no.nav.helsearbeidsgiver.utils.json.serializer.UuidSerializer
+import no.nav.helsearbeidsgiver.utils.json.toJson
+import no.nav.helsearbeidsgiver.utils.log.logger
+import no.nav.helsearbeidsgiver.utils.log.sikkerLogger
+import java.util.UUID
+
+data class LagreJournalpostIdMelding(
+    val eventName: EventName,
+    val behovType: BehovType,
+    val transaksjonId: UUID,
+    val inntektsmeldingType: InntektsmeldingType,
+    val inntektsmeldingId: UUID,
+    val journalpostId: String
+)
+
+// TODO erstatt med v1.Inntektsmelding.Type når mulig
+enum class InntektsmeldingType {
+    FORESPURT,
+    AAPEN
+}
+
+class LagreJournalpostIdRiver(
+    private val imRepo: InntektsmeldingRepository,
+    private val aapenImRepo: AapenImRepo
+) : ObjectRiver<LagreJournalpostIdMelding>() {
+
+    private val logger = logger()
+    private val sikkerLogger = sikkerLogger()
+
+    override fun les(json: Map<Key, JsonElement>): LagreJournalpostIdMelding? =
+        if (setOf(Key.DATA, Key.FAIL).any(json::containsKey)) {
+            null
+        } else {
+            val forespoerselId = Key.FORESPOERSEL_ID.lesOrNull(UuidSerializer, json)
```

Comment:
Det er vel for sent, men vil likevel nevne det igjen: Jeg ville heller hatt ett felt "ForespørselID" som ble benyttet overalt, da vil feilhåndtering, logging etc kun behøve å forholde seg til det ene feltet, nå har vi plutselig to forskjellige forespørselIDer. Å skille på en selvbestemt vs en vanlig IM kunne man gjort med et felt "type" ved opprettelse, heller enn å se på hva forespørselId-nøkkelen "heter". My 2 cents.

Reply:
Det er aldri for sent! Men jeg er ikke fan av å "lyve" og ha en selvbestemt ID i et felt som heter "forespoerselId". Det er kontraintuitivt. Men dine 2 cents er mine 2 cents. Jeg har allerede opprettet et typefelt i den nye IM-v1 for å gjøre skillet mellom selvbestemt og forespurt enklere å forholde seg til. Det var vel denne kodesnutten som førte til unnfangelsen av idéen.

Reply:
Jeg tenker imidlertid at sånn det er nå er _helt greit_, men så har man en enkel forbedringsmulighet når vi går over til den nye IM-klassen.